go/node/in_memory_node.go

```diff
@@ -0,0 +1,90 @@
+package node
+
+import (
+	"fmt"
+
+	"github.com/sanity-io/litter"
+
+	enclavecontainer "github.com/obscuronet/go-obscuro/go/enclave/container"
+	hostcontainer "github.com/obscuronet/go-obscuro/go/host/container"
+)
+
+type InMemNode struct {
+	cfg     *Config
+	enclave *enclavecontainer.EnclaveContainer
+	host    *hostcontainer.HostContainer
+}
+
+func NewInMemNode(cfg *Config) (Node, error) {
```

Comment:
We should return the `*InMemNode` here imo rather than `Node`, "Accept interfaces, return structs"

Reply:
Good point !